### DISCUSSION TITLE
Avalonia UI integration with Ranorex Spy

### DISCUSSION THREAD

**Discussion by reo-1963** [AvaloniaUI/Avalonia#9663] [Q&A] [answered]
Our testing department uses Ranorex for automated testing.
Within it is a utility called "Ranorex Spy" to identify the elements in the UI when the app runs.

I can build a simple DotNet 6 UI app in Visual Studio 2022, and the 10.2 version of Ranorex Spy can see it's UI elements.

But using the latest Avalonia NuGets (0.10.18 = Avalonia, Avalonia.Desktop, Avalonia.Diagnostics, Avalonia.ReactiveUI, XamlNameReferenceGenerator) on a simple "HelloWorld" MVVM program to display text on a Window, the Ranorex Spy
does not see the UI elements. (Ranorex Spy can see the top element of the window = "Title Bar", and the "Close", "Min", "Max",
but nothing inside.) I can use the "F12" to see the elements using the very nice "Avalonia Dev Tool".

Has anyone else figured out a way to make this work?

**Comment by maxkatz6** [comment, marked as answer]
> I can build a simple DotNet 6 UI app in Visual Studio 2022, and the 10.2 version of Ranorex Spy can see it's UI elements.

What exactly is "dotnet 6 UI app"? WPF or WinForms, I guess? They do support windows automation.

> But using the latest Avalonia NuGets (0.10.18

It is supported only from 11.0-preview versions.
See https://github.com/AvaloniaUI/Avalonia/issues/2157#issuecomment-1314705428
https://github.com/AvaloniaUI/Avalonia/issues/585

**Reply by reo-1963** [reply]
Thank you for replying.
I had a look at https://github.com/AvaloniaUI/Avalonia/issues/2157.
Looking forward to the release of 11.0
